tokio-timer/src/clock/clock.rs

```diff
@@ -0,0 +1,108 @@
+use clock::Now;
+use timer;
+
+use tokio_executor::Enter;
+
+use std::cell::Cell;
+use std::fmt;
+use std::sync::Arc;
+use std::time::Instant;
+
+/// TODO: Dox
+#[derive(Default, Clone)]
+pub struct Clock {
```

Comment:
Does this belong in `tokio-timer`?  What do you think about keeping this in a new `tokio-clock` or somesuch?  I can imagine tokio-dependent code benefiting from this primitive without using timers.

Reply:
Mostly, I would rather avoid introducing a new crate until there is a real case driving it. The two primary reasons to split the crates are that dependencies are different or you want to issue breaking changes at different times. The way I see it:

* `tokio-clock` would have the same dependencies as `tokio-timer`.
* If in the future, there is a reason to split them, this can be done in a backwards compatible way.

So, I would lean towards keeping it together for now until a solid case comes up.